File: app/src/main/java/edu/anbu/libraryofalexandria/AppDatabase.java
package edu.anbu.libraryofalexandria;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {Book.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {
    private static final String DATABASE_NAME = "BookDatabase.db";
    private static volatile AppDatabase instance;

    public static synchronized AppDatabase getInstance() {
        if(instance == null)
            instance = Room.databaseBuilder(LibraryOfAlexandria.getAppContext(), AppDatabase.class, DATABASE_NAME).build();
        return instance;
    }

    public abstract BookDao bookDao();
}

File: app/src/main/java/edu/anbu/libraryofalexandria/AppDatabase.kt
package edu.anbu.libraryofalexandria

import androidx.room.Database
import androidx.room.Room.databaseBuilder
import androidx.room.RoomDatabase

@Database(entities = [Book::class], version = 1)
abstract class AppDatabase : RoomDatabase() {
    abstract fun bookDao(): BookDao

    companion object {
        private const val DATABASE_NAME = "BookDatabase.db"

        @JvmStatic
        @get:Synchronized
        val instance: AppDatabase by lazy {
            databaseBuilder(
                LibraryOfAlexandria.appContext,
                AppDatabase::class.java,
                DATABASE_NAME
            ).build()
        }
    }
}